---
title: "R_assignment"
author: "Ping Kang"
date: "October 9, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
if (!require("tidyverse")) install.packages("tidyverse")
library(tidyverse)
if (!require("ggplot2")) install.packages("ggplot2")
library(ggplot2)
remove.packages("readr", lib="~/R/x86_64-pc-linux-gnu-library/3.5")
install.packages("readr")
remove.packages("tidyverse", lib="~/R/x86_64-pc-linux-gnu-library/3.5")
install.packages("tidyverse")
library("tidyverse")
# Import the files fang_et_al_genotypes.txt and snp_position.txt from class repository on github

fang <- read_tsv("https://raw.githubusercontent.com/pkangISU/R_assignment/master/fang_et_al_genotypes.txt")

snp <- read_tsv("https://raw.githubusercontent.com/pkangISU/R_assignment/master/snp_position.txt")

#check the number of columns in fang and snp
ncol(fang)
ncol(snp)

#check the number of rows in fang and snp
nrow(fang)
nrow(snp)

#check the first six rows of fang and snp
head(fang)
head(snp)

#check the last six rows of fang and snp
tail(fang)
tail(snp)

# pull out maize and teosinte files from fang

maize <- filter(fang, Group %in% c("ZMMIL","ZMMLR","ZMMMR"))
teosinte <- filter(fang, Group %in% c("ZMPBA","ZMPIL","ZMPJA"))
                   
ncol(maize)
nrow(maize)

ncol(teosinte)
nrow(teosinte)
#transpose maize and teosinte files by turning columns to rows
maize_t <- t(maize)
ncol(maize_t)
nrow(maize_t)

teosinte_t <- t(teosinte)
ncol(teosinte_t)
nrow(teosinte_t)

head(maize_t)

# pull out three columns from snp files
snp_3 <- select(snp, SNP_ID, Chromosome, Position)
ncol(snp_3)
nrow(snp_3)

str(maize_t)
attributes(maize_t)
is.list(maize_t)
is.vector(maize_t)
is.data.frame(maize_t)

#make sure that we have enough memory to run function merge
Sys.setenv('R_MAX_VSIZE'=100000000000)


#switch maize_t and snp_3 to data frame
maize_t <- as.data.frame(maize_t)
snp_3 <- as.data.frame(snp_3)


#view the maize_t and snp_3 files in table

View(maize_t)
View(snp_3)
```

#rename column 0 of maize_t as SNP_ID
maize_t <- cbind(SNP_ID =rownames(maize_t),maize_t)
colnames(maize_t) <- c("SNP_ID",1:1573)

#join maize and snp

maize_all <- merge.data.frame(snp_3,maize_t,by="SNP_ID")
ncol(maize_all)
nrow(maize_all)
head(maize_all)

#pull out each chromosome
maize_chr1 <- filter(maize_all,Chromosome == "1")
nrow(maize_chr1)
maize_chr2 <- filter(maize_all,Chromosome == "2")
maize_chr3 <- filter(maize_all,Chromosome == "3")
maize_chr4 <- filter(maize_all,Chromosome == "4")
maize_chr5 <- filter(maize_all,Chromosome == "5")
maize_chr6 <- filter(maize_all,Chromosome == "6")
maize_chr7 <- filter(maize_all,Chromosome == "7")
maize_chr8 <- filter(maize_all,Chromosome == "8")
maize_chr9 <- filter(maize_all,Chromosome == "9")
maize_chr10 <- filter(maize_all,Chromosome == "10")


#join teosinte and snp
View(teosinte_t)
teosinte_t <- as.data.frame(teosinte_t)
colnames(teosinte_t) <- c (1:975)
teosinte_t <- cbind(SNP_ID =rownames(teosinte_t),teosinte_t)

teosinte_all <- merge (snp_3,teosinte_t,by = "SNP_ID")
ncol(teosinte_all)
nrow(teosinte_all)
View(teosinte_all)

teosinte_chr1 <- filter(teosinte_all,Chromosome == "1")
nrow(teosinte_chr1)
teosinte_chr2 <- filter(teosinte_all,Chromosome == "2")
teosinte_chr3 <- filter(teosinte_all,Chromosome == "3")
teosinte_chr4 <- filter(teosinte_all,Chromosome == "4")
teosinte_chr5 <- filter(teosinte_all,Chromosome == "5")
teosinte_chr6 <- filter(teosinte_all,Chromosome == "6")
teosinte_chr7 <- filter(teosinte_all,Chromosome == "7")
teosinte_chr8 <- filter(teosinte_all,Chromosome == "8")
teosinte_chr9 <- filter(teosinte_all,Chromosome == "9")
teosinte_chr10 <- filter(teosinte_all,Chromosome == "10")
library(dplyr)
View(maize_chr1)
arrange(maize_chr1, as.numeric(Position))     
arrange(maize_chr1, desc(as.numeric(Position)),)


```
```{r}

```

```{r}

```

```{r}

```

```{r}

```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
